src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Comment.cs

```diff
@@ -113,14 +117,13 @@ private void WriteCommentIndented(ReadOnlySpan<char> value)
 
             Span<byte> output = _memory.Span;
 
-            if (_tokenType != JsonTokenType.None)
+            if (_tokenType != JsonTokenType.None || _commentAfterNoneOrPropertyName)
```

Comment:
optional suggestion: rename `_commentAfterNoneOrPropertyName` to `_forceNewLineAfterComment` - it took me a while to figure out what this code does - I think with the suggested name it makes it more obvious.